### PR TITLE
Add a warning if users pass --sentry or --sentry-nodes

### PR DESCRIPTION
**Pull request by tomaka**
cc #6762 

Let's not merge this without first informing everyone.

### CODE REVIEW

**Comment by tomaka** [comment]
bot rebase

File: client/network/src/service.rs [hunk 1]
```diff
@@ -180,6 +180,20 @@ impl<B: BlockT + 'static, H: ExHashT> NetworkWorker<B, H> {
...
 			}
 
+			let print_deprecated_message = match &params.role {
+				Role::Sentry { .. } => true,
+				Role::Authority { sentry_nodes } if !sentry_nodes.is_empty() => true,
+				_ => false,
+			};
+			if print_deprecated_message {
+				log::warn!(
+					"🙇 Sentry nodes are deprecated for the time being, and the `--sentry` and
```
Comment (mxinden) [inline client/network/src/service.rs:190]:
```suggestion
					"🙇 Sentry nodes are deprecated, and the `--sentry` and
```

File: client/network/src/service.rs [hunk 2]
```diff
@@ -180,6 +180,20 @@ impl<B: BlockT + 'static, H: ExHashT> NetworkWorker<B, H> {
...
+					`--sentry-nodes` CLI options will be removed in a future version. The Substrate
```
Comment (mxinden) [inline client/network/src/service.rs:191]:
```suggestion
					`--sentry-nodes` CLI options will eventually be removed in a future version. The Substrate
```

File: client/network/src/service.rs [hunk 3]
```diff
@@ -180,6 +180,20 @@ impl<B: BlockT + 'static, H: ExHashT> NetworkWorker<B, H> {
...
+					and Polkadot networking protocol require validators to be publicly-accessible.
+					Please do not block access to your validator nodes."
```
Comment (mxinden) [inline client/network/src/service.rs:193]:
```suggestion
					Please do not block access to your validator nodes.
					For details see https://github.com/paritytech/substrate/issues/6845.
```

**Comment by tomaka** [comment]
I suggest we merge now, so that this is in for the next release.

**Comment by tomaka** [comment]
bot merge

**Pull request merged**